[Actions][Docs] Actions and Connectors API Docs (#90974)
elastic/kibana: docs/api/actions-and-connectors/update.asciidoc

[[actions-and-connectors-api-update]]
=== Update action API
++++
<titleabbrev>Update action API</titleabbrev>
++++

> Update the attributes for an existing {kib} action

Reviewer: ```suggestion
Updates the attributes for an existing action.
```

[[actions-and-connectors-api-update-request]]
==== Request

`PUT <kibana host>:<port>/api/actions/action/<id>`

[[actions-and-connectors-api-update-params]]
==== Path parameters

`id`::

>   (Required, string) The ID of the action to update.

Reviewer: ```suggestion
  (Required, string) The ID of the action.
```

[[actions-and-connectors-api-update-request-body]]
==== Request body

`name`::

>   (Required, string) The updated display name for the action.

Reviewer: ```suggestion
  (Required, string) The new name of the action.
```

`config`::

>   (Required, object) The updated action configuration. Configuration properties vary depending on the action type. Some action types will have no configuration properties. See the specific action type to determine required fields.

Reviewer: ```suggestion
  (Required, object) The new action configuration. Configuration properties vary depending on the action type. For information about the configuration properties, refer to <<action-types,Action and connector types>>.
```

`secrets`::

>   (Required, object) The updated secrets configuration for the action. Secrets properties vary depending on the action type. Some action types will have no secrets properties. See the specific ation type to determine required fields.

Reviewer: ```suggestion
  (Required, object) The updated secrets configuration for the action. Secrets properties vary depending on the action type. For information about the secrets configuration properties, refer to <<action-types,Action and connector types>>.
```
